android/test-app-util.gradle

```diff
@@ -51,3 +51,25 @@ ext.findNodeModulesPath = { baseDir, packageName ->
 
     return null
 }
+
+ext.getFlipperVersion = { baseDir ->
+    def reactNativePath = findNodeModulesPath(baseDir, 'react-native')
+    def props = new Properties()
+    file("${reactNativePath}/template/android/gradle.properties").withInputStream {
+        props.load(it)
+    }
+
+    def flipperVersion = props.getProperty('FLIPPER_VERSION')
+    if (flipperVersion == null) {
+        // Current React Native version doesn't support Flipper
+        return null
+    }
+
+    // Prefer user specified Flipper version
+    if (project.hasProperty('FLIPPER_VERSION')) {
+        return project.getProperty('FLIPPER_VERSION')
+    }
+
+    // Use the recommended Flipper version
+    return flipperVersion
```

Comment:
We account for both 'FLIPPER_VERSION' being null or having a value before, so would we ever get here?

Reply:
There's a `FLIPPER_VERSION` which the user sets in their `gradle.properties`, and there's another one in `react-native` template that we use to detect whether the current version supports Flipper at all, as well as getting the recommended version. It's a bit confusing since we have to check that the value exists, and if so, check if the user has defined a version before returning the value we just got. Maybe it helps if I rename the variable to `recommendedFlipperVersion`?

Reply:
Aha, I see. I got confused there. Yeah maybe that would help the next person reading this.